### PR TITLE
Add modernize-use-nullptr check

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

**Review by DavidPoliakoff — APPROVED** [review]
Approved pending CI

File: core/src/Cuda/Kokkos_CudaSpace.cpp [hunk 1]
```diff
@@ -764,15 +765,16 @@ SharedAllocationRecord<Kokkos::CudaSpace, void>::get_record(void *alloc_ptr) {
   Header head;
 
   Header const *const head_cuda =
-      alloc_ptr ? Header::get_header(alloc_ptr) : (Header *)0;
+      alloc_ptr ? Header::get_header(alloc_ptr) : (Header *)nullptr;
```
Comment (dalg24) [inline core/src/Cuda/Kokkos_CudaSpace.cpp:768]: Do you actually need the cast with `nullptr`?

File: .clang-tidy [hunk 2]
```diff
@@ -1,2 +1,2 @@
-Checks: '-*,kokkos-*,modernize-use-using'
-FormatStyle: file
+Checks: '-*,kokkos-*,modernize-use-nullptr,kokkos-*,modernize-use-using'
```
Comment (masterleinad) [inline .clang-tidy:1]: This doesn't look good. I'll fix that quickly.

Comment (dalg24) [reply .clang-tidy:1]: thx

Comment (masterleinad) [reply .clang-tidy]: Fixed.

**Pull request merged**